ARExperiment/ARExperiment/SizeComparisonViewController.swift

```diff
@@ -84,20 +134,33 @@ extension SizeComparisonViewController: ARSCNViewDelegate {
         guard !anchor.isKind(of: ARPlaneAnchor.self) else {
             return nil
         }
-        guard let cube = arModel.createSceneNodeForAsset(currentCubeName,
-                                                         fileName: fileName,
-                                                         assetExtension: fileExtension),
-            let text = arModel.createSceneNodeForAsset(currentTextName,
-                                                       fileName: fileName,
-                                                       assetExtension: fileExtension) else {
-                                                        print("could not find node")
-                                                        return nil
+        guard let cube = arModel.createSceneNodeForAsset(currentCube.filename,
+                                                   fileName: fileName,
+                                                   assetExtension: fileExtension),
+            let text = arModel.createSceneNodeForAsset(currentCube.textFilename,
+                                                   fileName: fileName,
+                                                   assetExtension: fileExtension) else {
+                                                    print("could not find node")
+                                                    return nil
         }
         let node = SCNNode()
+        cube.scale = cube.scale.vectorScaled(to:self.currentScale.persentage)
+        text.scale = text.scale.vectorScaled(to:self.currentScale.persentage).vectorScaled(z:0.0)
         cube.position = SCNVector3Zero
-        text.position = SCNVector3Zero
+        text.position = cube.boundingBox.max.vectorScaled(to:self.currentScale.persentage)
         node.addChildNode(cube)
         node.addChildNode(text)
         return node
     }
 }
+
+extension SCNVector3 {
```

Comment:
There is a file for extensions and help functions, add that there since is possible we use it for other controllers

Reply:
I have moved them into a helper class